lib/unleash/client.rb

```diff
@@ -11,14 +12,17 @@ module Unleash
   class Client
     attr_accessor :fetcher_scheduled_executor, :metrics_scheduled_executor
 
+    # rubocop:disable Metrics/AbcSize
     def initialize(*opts)
       Unleash.configuration = Unleash::Configuration.new(*opts) unless opts.empty?
       Unleash.configuration.validate!
 
       Unleash.logger = Unleash.configuration.logger.clone
       Unleash.logger.level = Unleash.configuration.log_level
+      Unleash.engine = YggdrasilEngine.new
```

Comment:
I've chosen to allow the client to new up Ygg, other consumers do nil checks when they want to access it - that's the metrics handling code and the fetching code